File: src_backup/main/java/com/myudog/myulib/api/core/Color.java
package com.myudog.myulib.api.core;

/**
 * Shared color contract for systems that need named or RGB color values.
 */
public interface Color {
    int rgb();

    default float red() {
        return ((rgb() >> 16) & 0xFF) / 255.0f;
    }

    default float green() {
        return ((rgb() >> 8) & 0xFF) / 255.0f;
    }

    default float blue() {
        return (rgb() & 0xFF) / 255.0f;
    }
}

File: src/client/kotlin/com/myudog/myulib/client/api/ui/data/Color.kt
package com.myudog.myulib.client.api.ui.data

/**
 * [API] 輕量級色彩封裝 (使用 Inline Value Class 避免記憶體開銷)
 */
@JvmInline
value class Color(val argb: Int) {
    companion object {
        fun of(argb: Int) = Color(argb)
        fun fromHex(hex: String): Color {
            val cleanHex = hex.removePrefix("#")
            return Color(cleanHex.toLong(16).toInt())
        }
    }

    val alpha get() = (argb ushr 24) and 0xFF
    val red   get() = (argb ushr 16) and 0xFF
    val green get() = (argb ushr 8) and 0xFF
    val blue  get() = argb and 0xFF

    fun withAlpha(newAlpha: Int): Color {
        return Color((argb and 0x00FFFFFF) or (newAlpha shl 24))
    }
}